lib/matplotlib/axes/_axes.py

```diff
@@ -5303,9 +5315,14 @@ def reduce_C_function(C: array) -> float
             bins = np.sort(bins)
             accum = bins.searchsorted(accum)
 
+        if colorizer:
+            collection._set_colorizer_check_keywords(colorizer, cmap=cmap,
+                                                     norm=norm, vmin=vmin,
+                                                     vmax=vmax)
+        else:
+            collection.set_cmap(cmap)
+            collection.set_norm(norm)
         collection.set_array(accum)
-        collection.set_cmap(cmap)
-        collection.set_norm(norm)
         collection.set_alpha(alpha)
         collection._internal_update(kwargs)
         collection._scale_norm(norm, vmin, vmax)
```

Comment:
In my opinion the alpha keyword should always refer to the artist, because a primary use of the alpha keyword is to layer different artists in the same subplot, and you don't want the alpha to show up in the colorbar. With this logic, the `Colorizer` object should **not** have an alpha property, while `Artist` objects **should** have an alpha property (`Colorizer.to_rgba()` takes alpha as a keyword argument passed by the artist). 

I'm pretty sure this is how it is handled in the code as well.

In this specific case, `collections.Collection.set_alpha(alpha)` invokes `artist.Artist._set_alpha_for_array(alpha)`, which is independent of the `colorizer` framework.